common/settings/src/lib.rs

```diff
@@ -344,6 +353,16 @@ impl Settings {
         self.try_set_u64(key, val, false)
     }
 
+    pub fn get_enable_bloom_filter_index(&self) -> Result<u64> {
```

Comment:
Why we need a index settings here? For the user, I think they never know to enable them when they need it:)

Reply:
yeah, the same feeling here( the ultimate goal is user should not care about this). 
For this PR, maybe we should keep the bloom filter index as an option so that we can evaluate the performance impacts easier.
After the evaluation phase, let's abandon this setting. What do you think？